Integration/CityStateFromPostalCodeIntegration.php

```diff
@@ -124,13 +155,17 @@ public function doEnhancement(Lead &$lead)
             if (null !== $cityStatePostalCode) {
                 if (empty($leadCity) && !empty($cityStatePostalCode->getCity())) {
                     $this->logger->debug('CityStateFromPostalCode: Found city for lead '.$lead->getId());
-                    $lead->addUpdatedField('city', $cityStatePostalCode->getCity(), $leadCity);
+                    $lead->setCity($cityStatePostalCode->getCity());
                     $persist = true;
                 }
 
                 if (empty($leadState) && !empty($cityStatePostalCode->getStateProvince())) {
                     $this->logger->debug('CityStateFromPostalCode: Found state/province for lead '.$lead->getId());
-                    $lead->addUpdatedField('state', $cityStatePostalCode->getStateProvince(), $leadState);
+                    $stateProvince = $cityStatePostalCode->getStateProvince();
+                    $stateProvince = in_array($stateProvince, self::$US_STATES)
```

Comment:
To prevent confusion with $stateProvince below (they are opposites) can we rename one? 
Like $stateAbbr and $stateFull or something?

Reply:
$spLong is from the database
$spShort is the two character key from state array or spLong if non US